### PR TITLE
prefetch size fix for seting qos from rabbit

### PR DESCRIPTION
**Pull request by fersho**
## Subject

I am targeting this branch, because {reason}.

Closes #{put_issue_number_here}

## Changelog

```markdown
### Added
- Added prefetch size to configuration, because method setQos from rabbit doesnt accept null and it crashes

### Changed
Configuration
SonataNotificationExtension
AMQPBackend
### Deprecated

### Removed

### Fixed

### Security
```

### CODE REVIEW

File: src/Backend/AMQPBackend.php [hunk 1]
```diff
@@ -93,7 +98,7 @@ class AMQPBackend implements BackendInterface
      * @param string   $deadLetterRoutingKey
      * @param int|null $ttl
      */
-    public function __construct($exchange, $queue, $recover, $key, $deadLetterExchange = null, $deadLetterRoutingKey = null, $ttl = null, $prefetchCount = null)
+    public function __construct($exchange, $queue, $recover, $key, $deadLetterExchange = null, $deadLetterRoutingKey = null, $ttl = null, $prefetchCount = null, $prefetchSize = 0)
```
Comment (core23) [inline src/Backend/AMQPBackend.php]: Please add type hints here

Comment (fersho) [reply src/Backend/AMQPBackend.php]: done

File: src/Backend/AMQPBackend.php [hunk 2]
```diff
@@ -78,6 +78,11 @@ class AMQPBackend implements BackendInterface
      * @var int|null
      */
     private $prefetchCount;
+    
+    /**
+     * @var int|null
```
Comment (OskarStark) [inline src/Backend/AMQPBackend.php]: This one cannot be null

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -301,6 +307,12 @@ protected function getQueueNode()
                     ->max(65535)
                     ->defaultValue(null)
                 ->end()
+                ->integerNode('prefetch_size')
+                    ->info($prefetchSizeInfo)
+                    ->min(0)
+                    ->max(65535)
+                    ->defaultValue(null)
```
Comment (OskarStark) [inline src/DependencyInjection/Configuration.php]: Why not 0 like in the construct?

File: src/Backend/AMQPBackend.php (hunk 1 above)
Comment (greg0ire) [reply src/Backend/AMQPBackend.php]: Really? I don't see any type declaration :confused:

File: src/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -248,7 +248,13 @@ protected function getQueueNode()
 
 Defines the number of messages which will be delivered to the customer at a time.
 EOF;
+        $prefetchSizeInfo = <<<'EOF'
+Only used by RabbitMQ
+
+Defines the prefetch window size in octets.
```
Comment (greg0ire) [inline src/DependencyInjection/Configuration.php]: :fr:

**Pull request closed without merging**